File: 8 kyu/switch_case_bug_fixing_6.js
// Switch/Case - Bug Fixing #6

// oh no! timmy's evalObject function doesn't work. He uses Swithc/Cases to evaluate the given properties of an objec, can you fix timmy's function?

/*
using System;

public class Kata
{
  public static double EvalObject(double value1, double value2, char operation)
  {
    double result = 0;
    switch(operation)
    {
      case '+': result = value1 + value2;
      case '-': result = value1 - value2;
      case '/': result = value1 / value2;
      case '*': result = value1 * value2;
      case '%': result = value1 % value2;
      case '^': result = Math.Pow(value1, value2);
    }
    return result;
  }
}
*/




using System;

public class Kata
{
  public static double EvalObject(double value1, double value2, char operation)
  {
    double result = 0;
    switch(operation)
    {
      case '+': result = value1 + value2;
        break;
      case '-': result = value1 - value2;
        break;
      case '/': result = value1 / value2;
        break;
      case '*': result = value1 * value2;
        break;
      case '%': result = value1 % value2;
        break;
      case '^': result = Math.Pow(value1, value2);
        break;
    }
    return result;
  }
}

File: 8 kyu/switch_case_bug_fixing_6.py
# Switch/Case - Bug Fixing #6

# oh no! timmy's evalObject function doesn't work. He uses Swithc/Cases to evaluate the given properties of an objec, can you fix timmy's function?

"""def eval_object(v):
    match "operation":
        case "+":
            return v["a"] + v["b"]
        case "-":
            return v["a"] - v["b"]
        case "/":
            return v["a"] / v["b"]
        case "*":
            return v["a"] * v["b"]
        case "%":
            return v["a"] % v["b"]
        case "**":
            return v["a"] ** v["b"]
        case _:
            return 1
"""






def eval_object(v):
    if v["operation"] == "+":
        return v["a"] + v["b"]
    elif v["operation"] == "-":
        return v["a"] - v["b"]
    elif v["operation"] == "/":
        return v["a"] / v["b"]
    elif v["operation"] == "*":
        return v["a"] * v["b"]
    elif v["operation"] == "%":
        return v["a"] % v["b"]
    elif v["operation"] == "**":
        return v["a"] ** v["b"]
    





def eval_object(v):
    return {"+": v['a']+v['b'],
            "-": v['a']-v['b'],
            "/": v['a']/v['b'],
            "*": v['a']*v['b'],
            "%": v['a']%v['b'],
           "**": v['a']**v['b'], }.get(v['operation'])






def eval_object(v):
    return eval("{a}{operation}{b}".format(**v))






def eval_object(v):
    return eval(str(v['a'])+v['operation']+str(v['b']))